engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala

```diff
@@ -199,11 +216,21 @@ class EngineJobExecutionActor(jobKey: BackendJobDescriptorKey,
     context.actorOf(props, s"ejha_for_$jobDescriptor")
   }
 
-  def lookupCachedResult(jobDescriptor: BackendJobDescriptor, cacheResultId: MetaInfoId) = {
-    // TODO: Start up a backend job copying actor (if possible, otherwise just runJob). That should send back the BackendJobExecutionResponse
-    self ! FailedNonRetryableResponse(jobKey, new Exception("Call cache result copying not implemented!"), None)
-    // While the cache result is looked up, we wait for the response just like we were waiting for a Job to complete:
-    goto(RunningJob) using EmptyPartialCompletionData
+  def lookupCacheResult(jobDescriptor: BackendJobDescriptor, bjeaProps: Props, taskOutputs: Seq[TaskOutput], cacheResultId: MetaInfoId) = {
+    context.actorOf(FetchCachedResultsActor.props(cacheResultId, taskOutputs))
+    goto(PreparingCachedOutputs) using JobDescriptorData(jobDescriptor, bjeaProps)
+  }
+
+  def copyCacheHit(jobDescriptor: BackendJobDescriptor, cachedJobOutputs: JobOutputs) = {
+    val props = factory.cacheHitCopyingActorProps(jobDescriptor, initializationData, serviceRegistryActor, cachedJobOutputs)
+    val cacheHitCopyingActor = context.actorOf(props, buildCacheHitCopyingActorName(jobDescriptor))
+    // Send an ExecuteJobCommand to kick off processing.  The referenced actor will actually be a job execution actor if
+    // this backend doesn't implement caching.
+    cacheHitCopyingActor ! ExecuteJobCommand
+    context.parent ! JobRunning(jobDescriptor, cacheHitCopyingActor) //send the same message as runJob to WorkflowExecutionActor
+                                                                     //Workflow shouldn't care if it's execute or cache.
+                                                                     //switching mode occurs at the EJEA level?
+    goto(RunningJob) using PartialCompletionDataWithJobResults(cachedJobOutputs)
```

Comment:
Also, I don't think you want to use `PartialCompletionDataWithJobResults`. That would imply that the EJEA is no longer waiting for job results (it is!). Use the `EmptyPartialCompletionData` instead